src/OptimalControlUtilities/include/BipedalLocomotionControllers/OptimalControlUtilities/OptimizationProblemElements.h

```diff
@@ -0,0 +1,219 @@
+/**
+ * @file OptimizationProblemElements.h
+ * @authors Giulio Romualdi
+ * @copyright 2019 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTROLLERS_OPTIMAL_CONTROL_UTILITIES_OPTIMIZATION_PROBLEM_ELEMENTS_H
+#define BIPEDAL_LOCOMOTION_CONTROLLERS_OPTIMAL_CONTROL_UTILITIES_OPTIMIZATION_PROBLEM_ELEMENTS_H
+
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/Utils.h>
+#include <iDynTree/Core/VectorDynSize.h>
+
+#include <BipedalLocomotionControllers/OptimalControlUtilities/ControlProblemElements.h>
+#include <BipedalLocomotionControllers/OptimalControlUtilities/VariableHandler.h>
+
+namespace BipedalLocomotionControllers
+{
+namespace OptimalControlUtilities
+{
+/**
+ * Constraints class handles the equality and inequality constraint for a QP
+ * optimization problem. i.e. \f$ l \le A x \le u \f$ where
+ * \f$ A \f$ is the constraint matrix
+ * \f$ l \f$ and \f$ u \f$ are the lower and upper bound respectively
+ */
+class Constraints
+{
+    /**
+     * EqualityConstraint contains the equality constraints
+     */
+    typedef struct EqualityConstraint
+    {
+        CostFunctionOrEqualityConstraintElement* element; /**< Pointer to the constraint */
+        iDynTree::IndexRange indexRange; /**< Size and offset of the constraint */
+
+        /**
+         * Constructor
+         */
+        EqualityConstraint(CostFunctionOrEqualityConstraintElement* const element,
+                           const iDynTree::IndexRange& indexRange);
+    } EqualityConstraint;
```

Comment:
In C++ code, the following is not necessary: 
~~~
typedef struct A 
{
} A; 
~~~
is not necessary, because `A` already refers to `A`, differently from C where unless you have a typedef you need to refer to it as `struct A` .

Reply:
I didn't know. Thanks!